sectionproperties/pre/pre.py

```diff
@@ -505,7 +511,100 @@ def remove_duplicate_facets(self):
             self.geometry.facets.pop(idx)
 
 
-def create_mesh(points, facets, holes, control_points, mesh_sizes):
+def check_geometry(points, facets, holes, control_points, tolerance=1.0e-7):
+    """Check that holes and control points do not lie on a facet line.
+
+    :param points: List of points *(x, y)* defining the vertices of the cross-section
+    :type points: list[list[float, float]]
+    :param facets: List of point index pairs *(p1, p2)* defining the edges of the cross-section
+    :type points: list[list[int, int]]
+    :param holes: List of points *(x, y)* defining the locations of holes within the cross-section.
+        If there are no holes, provide an empty list [].
+    :type holes: list[list[float, float]]
+    :param control_points: A list of points *(x, y)* that define different regions of the
+        cross-section. A control point is an arbitrary point within a region enclosed by facets.
+    :type control_points: list[list[float, float]]
+    :param tolerance: minimum permissable distance from facet
+    :type tolerance: float
+
+    General method was adapted from:
+    https://stackoverflow.com/a/54442561
+    """
+    # make sure points is a numpy array
+    points = np.asarray(points)
+    check_points = np.array(holes + control_points)
+
+    # select the starting and ending points of each facet
+    facet_points = points[facets]
+    start = facet_points[:, 0, :]
+    end = facet_points[:, 1, :]
+
+    # create the vectors for each facet and normalize them to unit vectors
+    facet_vectors = end - start
+    lengths = np.sqrt(np.einsum('ij,ij->i', facet_vectors, facet_vectors))
+    if np.any(lengths == 0.0):
+        # if a length is zero, then a facet had the same starting and ending
+        # point
```

Comment:
I'm using a line length of 100 characters. This could probably be placed on line 546 for prettiness (I'm a bit OCD haha)

Reply:
and I'm OCD with using `black` but with line length of 79!  Haha.  But of course I should follow whatever the code style is of the library I'm committing too, I just wasn't sure what the section-properties default width was.  I'll update to make to 100 for this PR.